src/inference/ess.jl

```diff
@@ -155,6 +157,6 @@ function DynamicPPL.dot_tilde(rng, ctx::DefaultContext, sampler::Sampler{<:ESS},
     end
 end
 
-function DynamicPPL.dot_tilde(rng, ctx::DefaultContext, sampler::Sampler{<:ESS}, right, left, vi)
-    return DynamicPPL.dot_tilde(rng, ctx, SampleFromPrior(), right, left, vi)
+function DynamicPPL.dot_tilde(ctx::DefaultContext, sampler::Sampler{<:ESS}, right, left, vi)
+    return DynamicPPL.dot_tilde(ctx, SampleFromPrior(), right, left, vi)
```

Comment:
This was a bug. This is never actually hit and given the arguments I'm assuming it was intended to be a `dot_tilde_observe` rather than a `dot_tilde_assume` (which is the case when `rng` is passed, but because the signature doesn't match the rest of the `dot_tilde_assume`, it was never hit). The result was that  `ESS` didn't work for dotted observations.